### PR TITLE
[NCL-5812] Fix lazy loading workarounds

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java [hunk 1]
```diff
@@ -113,28 +113,19 @@ public ProductMilestoneProviderImpl(
...
 
         log.debug("Updating milestone for id: {}", id);
-        milestoneRestDb.setId(Integer.valueOf(id));
-
-        // make sure that user cannot set the 'endDate' via the REST API
-        // this should only be set after the release process is successful
-        milestoneRestDb.setEndDate(milestoneInDb.getEndDate());
-        // Make sure that user cannot change the product release of the milestone. This is set on release creation.
-        milestoneRestDb.setProductRelease(milestoneInDb.getProductRelease());
+        ((ProductMilestoneMapper) mapper).mergeEntity(restEntity, milestoneInDb);
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java:124]: Don't mind the cast here, it is for this prototype only. I would add the method to the root interface.

**Comment by vibe13** [comment]
@janinko yep I like it! it feels to me much cleaner and I also expect it to solve our lazy init workarounds (and make the caching layer happier). big +1

**Comment by janinko** [comment]
retest this please

**Comment by matejonnet** [comment]
I don't think merge is required, SpringRepository we are using is doing it behind the scene:
https://github.com/spring-projects/spring-data-jpa/blob/295664950d5f912450af5101a16a81c4fbf6e0ab/src/main/java/org/springframework/data/jpa/repository/support/SimpleJpaRepository.java#L552

File: facade/src/main/java/org/jboss/pnc/facade/providers/GroupConfigurationProviderImpl.java [hunk 2]
```diff
@@ -78,19 +78,13 @@ public GroupConfiguration getSpecific(String id) {
...
-        log.debug("Updating entity: " + restEntity);
-        BuildConfigurationSet dbEntity = repository.queryById(Integer.valueOf(id));
+    protected void validateBeforeUpdating(String id, GroupConfiguration restEntity) {
+        super.validateBeforeUpdating(id, restEntity);
+
+        BuildConfigurationSet dbEntity = findInDB(id);
         if (dbEntity.isArchived()) {
             throw new RepositoryViolationException("The Group Config " + id + " is already deleted.");
         }
-        // DO NOT REMOVE - Triggers the inizialization of LAZY collections (fixes NCL-5804)
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/GroupConfigurationProviderImpl.java:88]: MWAHAHA

Comment (vibe13) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GroupConfigurationProviderImpl.java:88]: clap clap

File: facade/src/main/java/org/jboss/pnc/facade/providers/ProductVersionProviderImpl.java [hunk 3]
```diff
@@ -152,9 +122,33 @@ protected void validateBeforeSaving(ProductVersion restEntity) {
...
+                        "Cannot change version due to having closed milestone. Product version id: " + id);
+            }
+        }
+    }
+
+    private void validateGroupConfigsBeforeUpdating(String id, ProductVersion restEntity)
+            throws InvalidEntityException, NumberFormatException, ConflictedEntryException {
         if (restEntity.getGroupConfigs() != null) {
             for (String groupConfigId : restEntity.getGroupConfigs().keySet()) {
+                log.warn("Checking group config: " + groupConfigId);
```
Comment (vibe13) [inline facade/src/main/java/org/jboss/pnc/facade/providers/ProductVersionProviderImpl.java:147]: are these leftovers logs for debugging? you may want to remove them?

**Review by vibe13 — APPROVED** [review]
Massive work @janinko , thanks for fixing this!

**Pull request merged**